mlir/python/requirements.txt

```diff
@@ -1,4 +1,4 @@
 numpy>=1.19.5, <=1.26
 pybind11>=2.9.0, <=2.10.3
-PyYAML>=5.3.1, <=6.0.1
-ml_dtypes   # provides several NumPy dtype extensions, including the bf16
\ No newline at end of file
+PyYAML>=5.4, <=6.0.1
```

Comment:
pybind explicitly says 2.9.0, maybe we can explicitly say 5.4.0 for PyYAML for consistency